### PR TITLE
pixel DQM@HLT

### PR DESCRIPTION
**Pull request by mtosi**
as PR https://github.com/cms-sw/cmssw/pull/18252

--finally--
we have the pixel@HLT DQM
this PR add the pixel@HLT DQM on both online and offline workflow

@fioriNTU
while testing this w/ runTheMatrix.py --limited
I saw that SiPixelPhase1TrackCluster was not checking the validity of the input collections,
not it is fixed
@fioriNTU 
while porting the code into 91x, I found that SiStripMonitorTrack is requring SiStripDigi, now
but we do not have them @HLT 
=> I updated the SiStripMonitorTrack as well

I also add the lumiMonitoring to the HLT offline DQM
(even if this module makes use of the lumi measurement from the scaler)

in addition, I fixed the strip DQM in the HLT online DQM as well
and add the monitoring of the HLT_ZeroBias_FirstCollisionAfterAbortGap
which is one of the calibration triggers for the tracker DPG (used for the strip APV gain)

finally, I add the client in the online DQM,
as now for both the tracking (iterative and GSF) and pixel

### CODE REVIEW

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Martin-Grunewald** [comment]
@mtosi 
Please address the errors in the tests of this PR!

**Comment by mtosi** [comment]
yes, martin
I'm working on it, but looking at the log
I'm really puzzled by the segmentation fault message
becasue --I think-- I do not make use of the DQMSiPixelPhase1Summary plugin ...
I'm re-running the 10024.0 workflow and I'll update this PR asap

**Comment by mtosi** [comment]
I was able to fix this issue by adding to the step_5HARVESTING.py

`del process.SiPixelPhase1Summary_Offline`

but I really do not understand how this PR triggers this issue,
because I do not think I touch / modify / add this module

this module as far --as I can see-- is called by 
`process.PixelOfflineDQMClientNoDataCertification`
which is called by
`process.DQMOffline_SecondStep_PreDPG`
by
`process.DQMOffline_SecondStep`
which belongs to
`process.dqmHarvesting`

could anybody help on this, please ?

**Comment by fioriNTU** [comment]
@mtosi I have run yesterday the workflow 10024 on  CMSSW_9_1_X_2017-04-09-1100/ and I didn't get any crash. I am puzzled too that the Phase1 Summary are still giving problems, @leggat fixed the initial crashes, maybe he can comment on this

**Comment by mtosi** [comment]
I added some cout
and it turns out the SiPixelPhase1Summary has the following
```
[SiPixelPhase1Summary::dqmEndJob] runOnEndJob_ ? YEAP
[SiPixelPhase1Summary::fillSummaries]
name: ADC
PixelPhase1/Phase1_MechanicalView/PXBarrel/Shell_mO/mean_adc_PXLayer_1
me: 0
```
and then it crashes

apparently the ME the code is expecting to have is not available
and the code is not protect against missing ME
I'll update the source and fix this issue,
but then it is not clear to me if this ME is expected to be available or not ....

please, @fioriNTU let me know
thanks !

**Comment by fioriNTU** [comment]
@mtosi actually we have a quite old PR fixing this issue https://github.com/cms-sw/cmssw/pull/18148
it ha not been merged yet in the Offline.

**Comment by mtosi** [comment]
uhmmm
I'm really confused on how w/o this fix the tests can run ....
anyhow, I already fixed the issues as well
it is doing the same but adding the LogWarning

please, let me know how to proceed

I think I'll need to ad the commit 
on the check of validity of the strip and pixel cluster collection in the `RecoTracker/MeasurementDet/plugins/MeasurementTrackerEventProducer.cc`
but before adding it, I'll wait for PR https://github.com/cms-sw/cmssw/pull/18252
checks and comments

hope it is fine
if anybody has further comments / suggestions
please let me know
thanks !

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Martin-Grunewald** [comment]
+1

**Comment by dmitrijus** [comment]
The conflict must be solved before signatures as any change will reset the signatures.

**Comment by dmitrijus** [comment]
+1

**Comment by mtosi** [comment]
sorry,
I do not understand the "The conflict must be solved before signatures as any change will reset the signatures."

could you please comment on this ?
thanks

**Comment by fwyzard** [comment]
The checks are saying that there is a conflict in file ` DQM/SiPixelPhase1Summary/src/SiPixelPhase1Summary.cc `, because it is modified by this PR but it has also been modified in the release since the point (commit) the branch for this PR was made.

**Comment by davidlange6** [comment]
-1
until merge conflict is resolved

**Comment by mtosi** [comment]
I think I did it, now

**Comment by fwyzard** [comment]
please test

**Comment by Martin-Grunewald** [comment]
+1

**Comment by mtosi** [comment]
is there anything preventing this PR to be integrated ?
thanks !

**Comment by Martin-Grunewald** [comment]
For example, the DQM signature is missing.

**Comment by mtosi** [comment]
@dmitrijus @vanbesien 
are there still open issues w/ this PR ?

**Comment by dmitrijus** [comment]
+1

File: DQM/SiStripMonitorTrack/src/SiStripMonitorTrack.cc [hunk 1]
```diff
@@ -34,13 +34,14 @@ SiStripMonitorTrack::SiStripMonitorTrack(const edm::ParameterSet& conf):
...
   TkHistoMap_On_ = conf.getParameter<bool>("TkHistoMap_On");
+  Digi_On_       = conf.getParameter<bool>("Digi_On");
 
   TrackProducer_ = conf_.getParameter<std::string>("TrackProducer");
   TrackLabel_    = conf_.getParameter<std::string>("TrackLabel");
 
   topFolderName_ = conf_.getParameter<std::string>("TopFolderName");
 
-  digiToken_       = consumes<edm::DetSetVector<SiStripDigi>> (  conf.getParameter<edm::InputTag>("ADCDigi_src") );
+  digiToken_       = mayConsume<edm::DetSetVector<SiStripDigi>> (  conf.getParameter<edm::InputTag>("ADCDigi_src") );
```
Comment (davidlange6) [inline DQM/SiStripMonitorTrack/src/SiStripMonitorTrack.cc]: Hi @mtosi - instead of a mayConsume, just add 

if (Digi_On_) digiToken_ = consumes<.....

**Comment by mtosi** [comment]
thanks @davidlange6

**Comment by dmitrijus** [comment]
+1

**Comment by davidlange6** [comment]
please test

**Pull request merged**